iothub/service/src/DirectMethod/Models/DirectMethodResponse.cs

```diff
@@ -18,13 +18,13 @@ public class DirectMethodResponse
         public int Status { get; internal set; }
 
         /// <summary>
-        /// Get the serialized JSON payload. May be null or empty.
+        /// Get the payload object. May be null or empty.
         /// </summary>
+        /// <remarks>
+        /// The payload can be null or Primitive type (e.g., String, Int)/Array/List/Map/custom type.
+        /// </remarks>
         [JsonIgnore]
-        public string Payload
-        {
-            get => (string)JsonPayload.Value;
-        }
+        public object Payload { get; set; }
```

Comment:
We should be looking at the code @abhipsaMisra has in the PnP preview branch and copy what we did there. We finely honed that API and wished to bring it here.

You can merge this in, but we're going to redo it, so you might want to look at that first and get us closer to the mark, at least.

Reply:
@drwill-ms  you mean this class: https://github.com/Azure/azure-iot-sdk-csharp/blob/previews/pnpApi/iothub/service/src/DirectMethod/CloudToDeviceMethodResult.cs

Reply:
Yes, for this class, but there are several other changes and associated classes.

Reply:
Will discuss the exact changes with @abhipsaMisra when she is back on Monday